### PR TITLE
EVG-2478 Refactor ec2 cost functions

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cloud/new_ec2.go [hunk 1]
```diff
@@ -544,10 +522,14 @@ func (m *ec2Manager) CostForDuration(h *host.Host, start, end time.Time) (float6
 		return 0, errors.Wrap(err, "error creating client")
 	}
 
-	if m.onDemand(h) {
-		return m.costForDurationOnDemand(h, start, end)
-	} else if m.spot(h) {
-		return m.costForDurationSpot(h, start, end)
+	t := timeRange{start, end}
```
Comment (tychoish) [inline cloud/new_ec2.go]: use named fields

File: cloud/new_ec2.go [hunk 2]
```diff
@@ -544,10 +522,14 @@ func (m *ec2Manager) CostForDuration(h *host.Host, start, end time.Time) (float6
...
+	ec2Cost, err := pkgCachingPriceFetcher.getEC2Cost(m.client, h, t)
+	if err != nil {
+		return 0, errors.Wrap(err, "error fetching ec2 cost")
+	}
+	ebsCost, err := pkgCachingPriceFetcher.getEBSCost(m.client, h, t)
```
Comment (tychoish) [inline cloud/new_ec2.go]: should probably access the global var via some mechanism to allow you to with more injection than this reference?

File: cloud/new_ec2_client.go [hunk 3]
```diff
@@ -45,6 +46,8 @@ type AWSClient interface {
 
 	// DescribeSpotPriceHistory is a wrapper for ec2.DescribeSpotPriceHistory.
 	DescribeSpotPriceHistory(*ec2.DescribeSpotPriceHistoryInput) (*ec2.DescribeSpotPriceHistoryOutput, error)
+
+	getInstanceInfo(string) (*ec2.Instance, error)
```
Comment (tychoish) [inline cloud/new_ec2_client.go]: mixing private and public methods in a single public interface is weird.

File: cloud/new_ec2.go (hunk 1 above)
Comment (bsamek) [reply cloud/new_ec2.go]: done

File: cloud/new_ec2.go (hunk 2 above)
Comment (bsamek) [reply cloud/new_ec2.go]: We pass the client, which is where the work happens, so I think this is fine.

File: cloud/new_ec2_client.go (hunk 3 above)
Comment (bsamek) [reply cloud/new_ec2_client.go]: done

File: cloud/new_ec2.go (hunk 2 above)
Comment (tychoish) [reply cloud/new_ec2.go]: I think you should either: make this just another type and drop the interface fantasy *or* make a factory somewhere to produce objects of this type.

Comment (bsamek) [reply cloud/new_ec2.go]: dropped the interface

**Pull request closed without merging**